operators/go/submariner_controller.go

```diff
@@ -0,0 +1,181 @@
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_submariner")
+
+// Add creates a new Submariner Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSubmariner{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("submariner-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Submariner
+	err = c.Watch(&source.Kind{Type: &submarinerv1alpha1.Submariner{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner Submariner
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &submarinerv1alpha1.Submariner{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileSubmariner implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileSubmariner{}
+
+// ReconcileSubmariner reconciles a Submariner object
+type ReconcileSubmariner struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Submariner object and makes changes based on the state read
+// and what is in the Submariner.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Submariner")
+
+	// Fetch the Submariner instance
+	instance := &submarinerv1alpha1.Submariner{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Create submariner-engine SA
```

Comment:
nit: delete this commented out block?

Reply:
I think the plan is to eventually uncomment this logic and use different SAs for Engine/Routeagent/Operator. I raised this Jira to track, after discussion with @mangelajo on a different comment here: https://jira.coreos.com/browse/HYCLD-249

Reply:
SAs are supposed to be created in CSVs, not in operators, AFAICT. That does mean it can be somewhat harder to deal with namespaces, but I don’t think that’s a problem here.

Reply:
Okay, it seems this is a non-feature, closing.